### PR TITLE
support for offset to do record pagination

### PR DESCRIPTION
**Pull request by sourcepirate**
Updates to searcher.rs to support extra arguments for python interface `search` function.

### CODE REVIEW

File: src/searcher.rs [hunk 1]
```diff
@@ -98,12 +98,13 @@ impl Searcher {
     /// Returns `SearchResult` object.
     ///
     /// Raises a ValueError if there was an error with the search.
-    #[args(limit = 10, count = true)]
+    #[args(limit = 10, offset = 0, count = true)]
```
Comment (poljar) [inline src/searcher.rs:101]: The new argument needs to be documented.

File: src/searcher.rs [hunk 2]
```diff
@@ -98,12 +98,13 @@ impl Searcher {
...
     fn search(
         &self,
         _py: Python,
         query: &Query,
         limit: usize,
+        offset: usize,
```
Comment (poljar) [inline src/searcher.rs:107]: Changing the order of the fields here is a breaking change, any reason why we don't want to put it at least after count?

**Review by poljar — CHANGES_REQUESTED** [review]
Generally looks good, needs some docs and if possible a test showing that this works as expected would be nice.

File: src/searcher.rs (hunk 2 above)
Comment (sourcepirate) [reply src/searcher.rs:107]: Got it!

**Comment by sourcepirate** [comment]
updated code based on the comments. Let me know if it is good enough.

File: src/searcher.rs (hunk 1 above)
Comment (sourcepirate) [reply src/searcher.rs:103]: done!

**Comment by sourcepirate** [comment]
@poljar any updates here ?

**Comment by poljar** [comment]
> @poljar any updates here ?

A bit patience please, I'll look at this later today. Taking a glance at this, we probably should check in the test that we got the correct document when using the offset.

**Comment by poljar** [comment]
I merged this but removed two commits that swapped the Tantivy version back and forth.

**Comment by sourcepirate** [comment]
Awesome! thanks @poljar

**Pull request closed without merging**